mmdet/models/backbones/resnet.py

```diff
@@ -473,8 +474,12 @@ def forward(self, x):
 
     def train(self, mode=True):
         super(ResNet, self).train(mode)
-        if mode and self.norm_eval:
-            for m in self.modules():
-                # trick: eval have effect on BatchNorm only
-                if isinstance(m, nn.BatchNorm2d):
-                    m.eval()
+        if mode:
+            self._freeze_stages()
+            if self.norm_eval:
+                for m in self.modules():
+                    # trick: eval have effect on BatchNorm only
+                    if isinstance(m, _BatchNorm):
+                        m.eval()
+        else:
+            convert_batchnorm(self)
```

Comment:
This may lead to a bug since `train()` and `eval()` are called alternately. If `eval()` is called and the SyncBN layers are converted to normal BN, in the next epoch it will not be converted back.

Reply:
The necessity of converting SyncBN to BN during testing is caused by current implementation of multi-gpu testing. The advantage that we can run multiple process on each GPU is not significant. Maybe we can discard the implementation and turn to ddp during testing.

Reply:
Thanks very much! That's would be much better than converting! `convert_batchnorm` will be deleted soon.